### ISSUE TITLE
Feature request: `beet diff`

### ISSUE THREAD

**Issue by jansol** [beetbox/beets#2536]
It would be nice to have a command for checking whether the beets db and the files are in sync. (I know, there's update and write with --pretend, but blindly entering either of those makes me feel uneasy every time)

**Comment by sampsyo** [comment]
Hmm; interesting suggestion. Would this do exactly the same thing as `write --pretend` or `update --pretend`, just with a nicer name? Or are you imaging some new functionality that neither does right now?

**Comment by jansol** [comment]
It could be functionally the same, just with improved output formatting.

**Comment by sampsyo** [comment]
What kind of formatting are you imagining? Perhaps we should just improve the formatting of those existing options.

**Comment by jansol** [comment]
I was thinking of something akin to `git diff`. The current formatting isn't that bad, but it could definitely show more clearly where each value is from. As it is currently I never remember which one is the value in the db an which one is in the file.

**Comment by sampsyo** [comment]
Hmm, I think I'm starting to understand. Can I convince you to try making a mockup of what you imagine the output would look like?

**Comment by jansol** [comment]
Can do, although not until exams are over, so probably some day next week.

**Comment by jansol** [comment]
Welp, totally forgot about this. Having thought about the issue on and off I've come to the conclusion that a simple note at the end of the listing could be enough. Something along the lines of
```
Running in pretend mode - no changes written to {files, database}
```
depending on which command was run. Additionally the change listing could show 📄 and 🗄 or some other fitting symbol on both sides of the arrow, but this is not going to work in everyone's terminal so I wouldn't rely solely on that. Like: 
```
key: C#m 🗄 -> 📄 None
```

**Comment by sampsyo** [comment]
OK! It sounds, though, like the actual output from one or the other command is what you want—you're just interested in greater clarity about what it represents?

**Comment by jansol** [comment]
Yeah, as discussed earlier, the update and write command in pretend mode output the relevant bits already, but it's a bit confusing as to which side is which and which one would be overwritten.

**Comment by sampsyo** [comment]
Got it; thanks! I've updated the title to reflect that.
